File: backend/src/algoritmos/arrays/aplanar_array/flatten.js
"use strict"

var a = ["elem1"],
    b = [2, 1, [3, [4, 5], 6], 7, [8]];

var flatten = function(l, source){
    if ( !source || !(source instanceof Array) ){ source = [] }
    for (var i = 0; i < l.length; i++){
        if (l[i] instanceof Array){ 
			flatten(l[i], source) 
		}
        else { source.push(l[i]) };
    };
    return source;
};

console.log( flatten(b, a) );

File: backend/src/algoritmos/arrays/aplanar_array/flatten.rb
#!/bin/ruby

a = ["elem1"]
b = [2, 1, [3, [4, 5], 6], 7, [8]]

def flatten(l, source=nil)
    if ( !source.instance_of?(Array) ); source = []; end;
    for i in l do
        if ( i.instance_of?(Array) )
            source = flatten(i, source)
        else
            source.push(i)
        end
    end
    return source
end

print flatten(b, a)
